test/TestCases/Generators/Linspace.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+  ==============================================================================
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+
+    Copyright 2022- by sonible GmbH.
+
+    This file is part of VCTR - Versatile Container Templates Reconceptualized.
+
+    VCTR is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License version 3
+    only, as published by the Free Software Foundation.
+
+    VCTR is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License version 3 for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    version 3 along with VCTR.  If not, see <https://www.gnu.org/licenses/>.
+  ==============================================================================
+*/
+
+#include <vctr_test_utils/vctr_test_common.h>
+
+namespace vctr::test
+{
+
+TEMPLATE_TEST_CASE ("Linspace (int, float, double)", "[VCTR][Generators][Linspace]", int, float, double)
+{
+    SECTION ("Length 1")
+    {
+        const size_t num = 1;
+        vctr::Vector<TestType> result (num);
+        result.fillLinspace (TestType (1), TestType (10), false);
+
+        REQUIRE_THAT (result, vctr::Equals (vctr::generators::Linspace::makeVector (TestType (1), TestType (10), num, false)));
+
+        REQUIRE (result.size() == 1);
+        REQUIRE (result[0] == TestType (1));
+    }
+
+    SECTION ("start = stop")
+    {
+        const size_t num = 10;
+        const auto value = TestType (3);
+
+        vctr::Vector<TestType> result (num);
+        result.fillLinspace (value, value, true);
+
+        REQUIRE_THAT (result, vctr::Equals (vctr::generators::Linspace::makeVector (value, value, num, true)));
+
+        // value should be repeated:
+        // >>> import numpy as np
+        // >>> np.linspace(start=3.0,stop=3.0, num=10, endpoint=True)
+        // array([3., 3., 3., 3., 3., 3., 3., 3., 3., 3.])
+        REQUIRE_THAT (result, vctr::Equals (vctr::Vector (result.size(), value)));
```

Comment:
`REQUIRE (result.allElementsEqual (value));` would be even more expressive in my opinion